### PR TITLE
Fix for memberships on group transfer

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Fixes the memberships after a group transfer, so that the access level is ascending through its descendants.
Project transfers and job creation is out of scope in this PR.
Fixes part of #226.

## Screenshots or screen recordings
Screenshot from 2023-10-19 14-04-19
Screenshot from 2023-10-19 14-05-15
Screenshot from 2023-10-19 14-06-25

## How to set up and validate locally
1. Create a new group called `Group A`.
2. Add user `user1@email.com` with `Maintainer` access to `Group A`.
3. Create a new group called `Group B`.
4. Add user `user1@email.com` with `Guest` access to `Group B`.
5. Transfer `Group B` to parent group `Group A`.
6. Verify that `user1@email.com` has changed to `Maintainer` in `Group B`.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/services/groups/transfer_service.rb [hunk 1]
```diff
@@ -19,8 +19,13 @@ def execute(new_namespace)
         raise TransferError, I18n.t('services.groups.transfer.namespace_group_exists')
       end
 
+      group_ancestor_members_user_ids = Member.for_namespace_and_ancestors(@group).map(&:user_id)
```
Comment (deepsidhu85) [inline app/services/groups/transfer_service.rb:22]: We should use `.select(:user_id)` over `map(&:user_id)` here. `select` will get just the `user_id` column for the table, whereas `map` gets all the columns and then gets the `user_id`s

You can try this in rails console to see the difference in the SQL statement

`group = Group.first`
`Member.for_namespace_and_ancestors(group).select(:user_id)`
`Member.for_namespace_and_ancestors(group).map(&:user_id)`

File: test/services/groups/transfer_service_test.rb [hunk 2]
```diff
@@ -11,18 +11,43 @@ def setup
     end
```
Comment (deepsidhu85) [inline test/services/groups/transfer_service_test.rb:12]: Can I get you to add a test for this  scenario

`Group A` -> member has `MAINTAINER` access
`Group A Subgroup A` -> above owner is added directly and member has `OWNER` access
`Group B` -> member has `GUEST` access

Transfer `Group B` under `Group A Subgroup A`

Ensure that in `Group A` the member still has `MAINTAINER` access and that only `Group A Subgroup A` and `Group B` has the member's access level been updated to `OWNER`

I would suggest adding all new groups, and members to the fixtures for this

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good! Just have a few comments below.

File: app/services/groups/transfer_service.rb (hunk 1 above)
Comment (ksierks) [reply app/services/groups/transfer_service.rb:22]: Ok. 

Screenshot from 2023-10-26 14-18-39

Comment (deepsidhu85) [reply app/services/groups/transfer_service.rb]: You can just use the result of `.select(:user_id)`. Should not have to use pluck

File: test/services/groups/transfer_service_test.rb (hunk 2 above)
Comment (ksierks) [reply test/services/groups/transfer_service_test.rb:12]: I changed `Group A` to `Group 9` and `Group B` to `Group 10`. Please see 43d8a91172044d228939828cc9effe16cf190e61.

File: app/models/member.rb [hunk 3]
```diff
@@ -19,7 +19,7 @@ class Member < ApplicationRecord # rubocop:disable Metrics/ClassLength
   validate :validate_namespace
   validate :higher_access_level_than_group
 
-  after_update :update_descedant_memberships
+  after_update :update_descendant_memberships
```
[the remark this answers is not in the document]
Comment (ksierks) [reply app/models/member.rb:22]: Yup!

File: app/services/groups/transfer_service.rb (hunk 1 above)
Comment (ksierks) [reply app/services/groups/transfer_service.rb]: Please see d72ff36bb8eaca2c0a460fb9ace62ec6f1447beb.

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Pull request merged**